sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -859,6 +859,22 @@ class Analyzer(
       // rule: ResolveDeserializer.
       case plan if containsDeserializer(plan.expressions) => plan
 
+      case u @ UnresolvedSubqueryColumnAliases(columnNames, child) if child.resolved =>
```

Comment:
The last question. What is the reason why we do not have a separate analyzer rule?